chain/client/src/sync/state.rs

```diff
@@ -94,15 +97,18 @@ impl PendingRequestStatus {
     }
 }
 
+pub enum StateSyncFileDownloadResult {
+    StateHeader { header_length: u64, header: ShardStateSyncResponseHeader },
+    StatePart { part_length: u64 },
+}
+
 /// Signals that a state part was downloaded and saved to RocksDB.
 /// Or failed to do so.
-pub struct StateSyncGetPartResult {
+pub struct StateSyncGetFileResult {
     sync_hash: CryptoHash,
     shard_id: ShardId,
-    part_id: PartId,
-    /// Either the length of state part, or an error describing where the
-    /// process failed.
-    part_result: Result<u64, String>,
+    part_id: Option<PartId>,
+    result: Result<StateSyncFileDownloadResult, String>,
```

Comment:
Can you move the part id to the inside of StateSyncFileDownloadResult::StatePart?

Reply:
No, because in case of error, we would not know which part failed.